File: backend/src/main/java/dev/tripdraw/domain/Point.java
package dev.tripdraw.domain;

import static lombok.AccessLevel.PROTECTED;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = PROTECTED)
@Entity
public class Point extends BaseEntity {

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    @Column(nullable = false)
    private LocalDateTime recordedAt;

    @Builder
    public Point(Double latitude, Double longitude, LocalDateTime recordedAt) {
        this.latitude = latitude;
        this.longitude = longitude;
        this.recordedAt = recordedAt;
    }
}

File: android/domain/src/main/java/com/teamtripdraw/android/domain/model/point/Point.kt
package com.teamtripdraw.android.domain.model.point

import java.time.LocalDateTime

data class Point(
    val pointId: Long,
    val latitude: Double,
    val longitude: Double,
    val hasPost: Boolean,
    val recordedAt: LocalDateTime,
) {
    companion object {
        const val NULL_SUBSTITUTE_ID = -1L
        const val NULL_LATITUDE = (-100).toDouble() // latitude는 -90 ~ 90 범위의 값을 가집니다.
        const val NULL_LONGITUDE = (-200F).toDouble() // longitude는 -180 ~ 180 범위의 값을 가집니다.
    }
}
